cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,315 @@
+	command.Flags().BoolVar(&opts.isURL, "url", false, "install plugin from an HTTPS URL. The timeout of the download HTTPS request is set to 10 minutes")
+	command.Flags().StringVar(&opts.inputChecksum, "sha256sum", "", "must match SHA256 of the plugin source, required when \"--url\" flag is set")
+	command.Flags().BoolVar(&opts.force, "force", false, "force the installation of the plugin")
+	command.MarkFlagsMutuallyExclusive("file", "url")
+	command.MarkFlagsOneRequired("file", "url")
+	return command
+}
+
+func install(command *cobra.Command, opts *pluginInstallOpts) error {
+	// set log level
+	ctx := opts.LoggingFlagOpts.InitializeLogger(command.Context())
+	// core process
+	switch opts.pluginSourceType {
+	case notationplugin.PluginSourceTypeFile:
+		if err := installPlugin(ctx, opts.pluginSource, opts.inputChecksum, opts.force); err != nil {
+			return fmt.Errorf("plugin installation failed: %w", err)
+		}
+		return nil
+	case notationplugin.PluginSourceTypeURL:
+		if opts.inputChecksum == "" {
+			return errors.New("installing from URL requires non-empty SHA256 checksum of the plugin source")
+		}
+		pluginURL, err := url.Parse(opts.pluginSource)
+		if err != nil {
+			return fmt.Errorf("failed to parse plugin download URL %s with error: %w", pluginURL, err)
+		}
+		if pluginURL.Scheme != "https" {
+			return fmt.Errorf("failed to download plugin from URL: only the HTTPS scheme is supported, but got %s", pluginURL.Scheme)
+		}
+		tmpFile, err := os.CreateTemp("", notationPluginDownloadTmpFile)
+		if err != nil {
+			return fmt.Errorf("failed to create notationPluginDownloadTmpFile: %w", err)
+		}
+		defer os.Remove(tmpFile.Name())
+		defer tmpFile.Close()
+		fmt.Printf("Downloading plugin from %s\n", opts.pluginSource)
+		err = notationplugin.DownloadPluginFromURL(ctx, opts.pluginSource, tmpFile)
+		if err != nil {
+			return fmt.Errorf("failed to download plugin from URL %s with error: %w", opts.pluginSource, err)
+		}
+		fmt.Println("Download completed")
+		if err := installPlugin(ctx, tmpFile.Name(), opts.inputChecksum, opts.force); err != nil {
+			return fmt.Errorf("plugin installation failed: %w", err)
+		}
+		return nil
+	default:
+		return errors.New("plugin installation failed: unknown plugin source type")
+	}
+}
+
+// installPlugin installs the plugin given plugin source path
+func installPlugin(ctx context.Context, inputPath string, inputChecksum string, force bool) error {
+	// sanity check
+	inputFileStat, err := os.Stat(inputPath)
+	if err != nil {
+		return err
+	}
+	if !inputFileStat.Mode().IsRegular() {
+		return fmt.Errorf("%s is not a valid file", inputPath)
+	}
+	// checksum check
+	if inputChecksum != "" {
+		if err := osutil.ValidateSHA256Sum(inputPath, inputChecksum); err != nil {
+			return err
+		}
+	}
+	// install the plugin based on file type
+	fileType, err := osutil.DetectFileType(inputPath)
+	if err != nil {
+		return err
+	}
+	switch fileType {
+	case notationplugin.MediaTypeZip:
```

Comment:
Can we please add test for [zipbomb](https://en.wikipedia.org/wiki/Zip_bomb), [zipslip](https://security.snyk.io/research/zip-slip-vulnerability), ZIP filename spoofing ?

Reply:
One thing to point out is that according to notation's threat model: https://github.com/notaryproject/specifications/blob/main/threatmodels/notation-threatmodel.md

malicious plugin is mitigated by `Verify plugin package before installation or plugin is installed by authorized users`.